defaults/templates/the-build/build.xml

```diff
@@ -1,25 +1,78 @@
-<project name="@projectname@" default="build">
-    <!-- Run with: vendor/bin/phing -Dbuild.env=vagrant -->
-    <!-- For a list of commands, try: vendor/bin/phing -Dbuild.env=vagrant -l -->
+<?xml version="1.0"?>
+
+<!--
+  @file build.xml
+  Build targets for this project. This file should be customized for your project.
+
+  For a full list of targets, try:
+    phing -l
 
+  @see https://github.com/palantirnet/the-build
+  -->
+
+<project name="@projectname@" default="build">
 
+    <!-- Add property loading and tasks from palantirnet/the-build. -->
     <import file="vendor/palantirnet/the-build/tasks/the-build.xml" />
 
 
-    <!-- Make these commands available by default. -->
+    <!-- Make these additional targets available by default. -->
     <import file="vendor/palantirnet/the-build/tasks/drupal.xml" />
+
+
+    <!-- Additional optional targets. -->
     <import file="vendor/palantirnet/the-build/tasks/artifact.xml" />
+    <import file="vendor/palantirnet/the-build/tasks/acquia.xml" />
 
 
     <!-- Default target: build -->
     <target name="build" description="Build the application.">
-        <phingcall target="drupal-build" />
+        <!-- Create the Drupal custom code directories. -->
+        <foreach list="${drupal.create_dirs}" param="dir" target="mkdir" />
+
+        <!-- Set permissions on the sites subdirectory. -->
+        <mkdir dir="${drupal.root}/sites/${drupal.sites_subdir}" />
+        <chmod file="${drupal.root}/sites/${drupal.sites_subdir}" mode="750" />
+
+        <!-- Create the public files directory and make it world writable. -->
+        <mkdir dir="${drupal.root}/${drupal.settings.file_public_path}" />
+        <chmod file="${drupal.root}/${drupal.settings.file_public_path}" mode="744" />
+
+        <!-- Create the private files directory and make it private. -->
+        <mkdir dir="${drupal.root}/${drupal.settings.file_private_path}" />
+        <chmod file="${drupal.root}/${drupal.settings.file_private_path}" mode="640" />
+
+        <!-- Copy the settings.build.php and services.build.php templates into place. -->
```

Comment:
I'm not sure if it's happening in this section, but I was testing an artifact build with Pantheon and noticed `settings.build.php` is getting added, which breaks the site on Pantheon.  We're calling this from `artifact-build` with `inheritAll="false"` which makes me think maybe `build.drupal.settings` et. al. are getting reset to defaults.  Do we need to pass them in from `artifact-build`?

Reply:
Along those lines, it might be good to add a `pantheon.yml` that sets `web_docroot=true`.  I'm not sure where it's best to add, though.  Maybe in the installer if Pantheon is selected as the hosting platform?

Reply:
Ah, interesting. I would love to add Pantheon and Platform config but I was holding off because it would be easy to rabbit hole -- but that's exactly where I would add it, in the installer.

I do think the first bit (about settings.build.php being added to a pantheon artifact) needs to be fixed before this can be merged, so I'll check that out.

Reply:
Can you give me more details on this issue? What I find is that when I run `phing build -Dbuild.env=pantheon`, the-build does create a `settings.build.php` file, but it's a limited version of the settings, customized for Pantheon:

```php
<?php

/**
 * @file
 * Drupal settings file template for use on Pantheon environments.
 */

$config_directories = [];
$config_directories[CONFIG_SYNC_DIRECTORY] = '../config/sites/default';
```

Is it this file that causes issues? What issues does it cause?

Reply:
Ohhh, wait, in the artifact process. Hang on...

Reply:
... and I went ahead and added Pantheon and Platform.sh hosting config files.